actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -321,23 +321,26 @@ def non_kwarg_request_warning
         end
 
         # Performs the actual request.
-        def process(method, path, params: nil, headers: nil, env: nil, xhr: false)
+        def process(method, path, params: nil, headers: nil, env: nil, xhr: false, as: nil)
+          request_encoder = RequestEncoder.encoder(as)
+          location = URI.parse(path)
```

Comment:
Should probably do a test for how much of an impact this has on our integration test speed.